### PR TITLE
Switch from tslint to eslint

### PR DESCRIPTION
**Pull request by nlisgo**
Closes: https://github.com/libero/reviewer-submission/issues/12

### CODE REVIEW

File: src/packages/submission/submission.repository.ts [hunk 1]
```diff
@@ -4,16 +4,16 @@ import { uuidType } from 'typesafe-uuid';
...
+    findAll(): Promise<Submission[]>;
+    findById(id: SubmissionId): Promise<Option<Submission>>;
+    save(subm: Submission): Promise<Submission>;
+    delete(id: SubmissionId): Promise<boolean>;
 }
 
-// I'm treating ISubmission as a DTO for submission
+// I'm treating Submission as a DTO for Submission (./submission.entity)
 
-export interface ISubmission {
```
Comment (nlisgo) [inline src/packages/submission/submission.repository.ts]: The specific error was: `Interface name must not be prefixed with "I"`

File: src/packages/survey/survey-response.repository.ts [hunk 2]
```diff
@@ -3,30 +3,29 @@ import { Answer } from './answer';
...
-// tslint:disable:max-classes-per-file
 export class SurveyId extends uuidType<'SurveyId'>() {}
 export class SurveyResponseId extends uuidType<'SurveyResponseId'>() {}
 
 export interface SurveyResponseRepository {
-  save(surveyResponse: ISurveyResponse): Promise<ISurveyResponse>;
+    save(surveyResponse: SurveyResponse): Promise<SurveyResponse>;
 }
 
-export interface ISurveyResponse {
```
Comment (nlisgo) [inline src/packages/survey/survey-response.repository.ts]: The specific error was: `Interface name must not be prefixed with "I"`

File: src/packages/submission/submission.repository.ts (hunk 1 above)
Comment (diversemix) [reply src/packages/submission/submission.repository.ts]: Let's change the `prefix` setting in the linter

File: .eslintrc.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+    'plugin:@typescript-eslint/recommended',  // Uses the recommended rules from @typescript-eslint/eslint-plugin
+    'prettier/@typescript-eslint',  // Uses eslint-config-prettier to disable ESLint rules from @typescript-eslint/eslint-plugin that would conflict with prettier
+    'plugin:prettier/recommended',  // Enables eslint-plugin-prettier and displays prettier errors as ESLint errors. Make sure this is always the last configuration in the extends array.
+  ],
+  parserOptions:  {
+    ecmaVersion:  2018,  // Allows for the parsing of modern ECMAScript features
+    sourceType:  'module',  // Allows for the use of imports
+  },
+  rules: {
+    '@typescript-eslint/interface-name-prefix': 'off'
```
Comment (nlisgo) [inline .eslintrc.js]: @diversemix with this can we resolve your request for changes?

Comment (diversemix) [reply .eslintrc.js]: Checking...

**Pull request merged**